go-controller/pkg/config/config.go

```diff
@@ -1743,10 +1776,29 @@ func completeGatewayConfig(allSubnets *configSubnets) error {
 	if err != nil || !utilnet.IsIPv6(v6IP) {
 		return fmt.Errorf("invalid gateway v6 join subnet specified, subnet: %s: error: %v", Gateway.V6JoinSubnet, err)
 	}
-
 	allSubnets.append(configSubnetJoin, v4JoinCIDR)
 	allSubnets.append(configSubnetJoin, v6JoinCIDR)
 
+	//validate v4 and v6 masquerade subnets
+	v4MasqueradeIP, v4MasqueradeCIDR, err := net.ParseCIDR(Gateway.V4MasqueradeSubnet)
+	if err != nil || utilnet.IsIPv6(v4MasqueradeIP) {
+		return fmt.Errorf("invalid gateway v4 masquerade subnet specified, subnet: %s: error: %v", Gateway.V4MasqueradeSubnet, err)
+	}
+	if err = allocateV4MasqueradeIPs(v4MasqueradeIP, masqueradeIPs); err != nil {
```

Comment:
Is there a reason you replaced `v4MasqueradeCIDR.IP` with `v4MasqueradeIP` in the latest changes?

Reply:
must have missed that change when i rolled in the reflog to make some changes, corrected it